thetis/coordsys.py

```diff
@@ -0,0 +1,97 @@
+"""
+Generic methods for converting data between different spatial coordinate systems.
+Uses pyproj library.
+"""
+
+import pyproj
+import numpy as np
+
+# NOTE this adds pyproj as dependencency
+# TODO generalize; define unique names for commonly-used systems
+
+
+UTM_ZONE10 = pyproj.Proj(
+    proj='utm',
+    zone=10,
+    datum='NAD83',
+    units='m',
+    errcheck=True)
+SPCS_N_OR = pyproj.Proj(init='nad27:3601', errcheck=True)
+LL_WGS84 = pyproj.Proj(proj='latlong', datum='WGS84', errcheck=True)
+LL_WO = pyproj.Proj(proj='latlong', nadgrids='WO', errcheck=True)  # HARN
+
+
+def convertCoords(x, y, fromSys, toSys):
+    """Converts x,y from fromSys to toSys."""
+    if isinstance(x, np.ndarray):
+        # proj may give wrong results if nans in the arrays
+        lon = np.ones_like(x) * np.nan
+        lat = np.ones_like(y) * np.nan
+        goodIx = np.logical_and(np.isfinite(x), np.isfinite(y))
+        lon[goodIx], lat[goodIx] = pyproj.transform(
+            fromSys, toSys, x[goodIx], y[goodIx])
+    else:
+        lon, lat = pyproj.transform(fromSys, toSys, x, y)
+    return lon, lat
+
+
+def getVectorRotationMatrix(lon, lat, target_csys, source_csys=None,
+                            delta=None):
+    """
+    Estimate rotation matrix that converts vectors from source_csys to
+    target_csys at (lon, lat) location.
+
+
+    A vector can defined at (lon, lat) can then be transformed as
+
+    .. code-block:: python
+
+        R, theta = getVectorRotationMatrix(lon, lat, target_csys)
+        v_ll = numpy.array([[v_lon], [v_lat]])
+        v_csys = numpy.matmul(R, v_ll)
+        v_x, v_y = v_csys
+
+    """
+    if source_csys is None:
+        source_csys = LL_WO
```

Comment:
Is this a sensible default? Also this default could be set in the argument list? (so it's more easily discoverable)